tests/generator-Tests/Unit-Tests/CodeGeneratorExpectedResults/Common/WriteInterfaceProperties.txt

```diff
@@ -5,7 +5,7 @@ int Count {
 	[Register ("set_Count", "(I)V", "Getset_Count_IHandler:java.code.IMyInterfaceInvoker, ")] set;
 }
 
-java.lang.String Key {
+string Key {
```

Comment:
This change concerns me: why did it occur?  For that matter, how was it `java.lang.String` in the *first* place?

Fortunately this doesn't trigger any API breakage in `Mono.Android.dll`, as per https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=3648625&view=results

I just find this really weird.  What change to `generator` is triggering this?  Was it deliberate?  What's the rationale?

I don't see this mentioned in the commit message.

Reply:
The change is that previously we were writing this type from `property.Type`, which is `Setter != null ? Setter.Parameters [0].Type : Getter.ReturnType`.  

https://github.com/xamarin/java.interop/blob/master/tools/generator/Java.Interop.Tools.Generator.CodeGeneration/CodeGenerator.cs#L1690

This was refactored and combined with other places that were writing `property.Getter.ReturnType` to a new method, which always prefers `Getter.ReturnType`:

https://github.com/xamarin/java.interop/pull/563/files#diff-18357c6193fe262963d21bb1d0bab9feR99-R105

The test is a unit test and does not go through the full pipeline of resolving all the types, thus its `Setter.Parameters [0].Type` isn't changing from `java.lang.String` to `string`, but its `Getter.ReturnType` is getting resolved to `string`.

In the real world, `Setter.Parameters [0].Type` and `Getter.ReturnType` should always refer to the same symbol, so this should not affect real code.